src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -27,12 +31,18 @@
 
 public class HologramProjector extends SlimefunItem {
 
-    private static final String OFFSET_PARAMETER = "offset";
+    private static final String X_OFFSET_PARAMETER = "x_offset";
+    private static final String Y_OFFSET_PARAMETER = "y_offset";
+    private static final String Z_OFFSET_PARAMETER = "z_offset";
+    private static final int[] BORDER = {1, 3, 5, 7, 8, 10, 12, 14, 16, 17};
+
+    private static final ItemSetting<Double> maxDistance = new ItemSetting<>("max-horizontal-offset", 5.0);
```

Comment:
I think a ranged Setting would be quite useful here, I quickly threw together one for you:
https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/api/items/settings/DoubleRangeSetting.java

(Yes, I literally just copied the file on my phone and replaced int with double, deal with it.